### PR TITLE
Modify the stratos pipeline to use go version 1.18.3

### PR DESCRIPTION
**Pull request by rcgottlieb**
## Summary
There are findings against several versions of the language go. Stratos is currently using one of those versions: 1.12. By bringing the codebase to the most current version we are in compliance. This PR merges code changing the pipeline to separate out the build task and to use go 1.18.3

## Description
We are in compliance after this and the Stratos PR are merged based on the following:
1. We updated the pipeline to separate out the task to build the go code, allowing us to change the version of go to 1.18.3
2. In moving the steps out of the codebase and into the pipeline, we were able to make the build work with a build container

## Motivation and Context
We are currently out of compliance using go version 1.12 in the Stratos codebase.

## How Has This Been Tested?
By updating the pipeline in this codebase to use go version 1.18.3, and testing with Stratos branch apb/rollback-stratos.

First observing that the dashboard was deployed, and then exercising some of the UI functionality to verify it was functioning as expected.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Docs update
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have followed the guidelines in CONTRIBUTING.md, including the required formatting of the commit message

## Security Considerations
Supports the update of go to 1.18.3 to be in compliance.

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -266,7 +308,7 @@ resources:
   icon: github-circle
   source:
     uri: https://github.com/cloud-gov/stratos
-    tag_filter: '4.1.0'
+    branch: apb/rollback-stratos
```
Comment (apburnes) [inline ci/pipeline.yml:311]: Let's update this to a new tag.

File: compile [hunk 2]
```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
```
Comment (apburnes) [inline compile:1]: Do we need this file? I think we can delete it.

File: ci/pipeline.yml [hunk 3]
```diff
@@ -88,13 +125,16 @@ jobs:
...
-      path: precompiled/
+      path: compiled
       current_app_name: stratos
-      manifest: precompiled/manifest.yml
+      manifest: compiled/manifest.yml
       show_app_log: true
       vars:
         route: ((dev-cf-route))
       environment_variables:
+        # NODE_VERSION: "15.15"
```
Comment (apburnes) [inline ci/pipeline.yml:135]:
```suggestion
```

**Review by apburnes — COMMENTED** [review]
Added a few suggestions for things we don't need.

File: ci/pipeline.yml (hunk 1 above)
Comment (rcgottlieb) [reply ci/pipeline.yml]: Modified to use the master branch for Stratos now that the PR has been merged for Stratos.

File: compile (hunk 2 above)
Comment (rcgottlieb) [reply compile:1]: No we don't need it. Debugging is pipeline based. If we end up needing it, we can either resurrect it from git, or create a new one that may end up being more relevant.

File: ci/pipeline.yml [hunk 4]
```diff
@@ -138,13 +178,15 @@ jobs:
...
-      path: precompiled/
+      path: compiled/
       current_app_name: stratos
-      manifest: precompiled/manifest.yml
+      manifest: compiled/manifest.yml
       show_app_log: true
       vars:
         route: ((staging-cf-route))
       environment_variables:
+        GO_VERSION: "1.18.1"
```
[the remark this answers is not in the document]
Comment (rcgottlieb) [reply ci/pipeline.yml:186]: Updated per below comment about using 1.18.3

File: ci/pipeline.yml (hunk 1 above)
Comment (apburnes) [reply ci/pipeline.yml:310]: Let's update this to `master` since the stratos PR has been merged.

File: ci/pipeline.yml [hunk 5]
```diff
@@ -266,7 +306,8 @@ resources:
   icon: github-circle
   source:
     uri: https://github.com/cloud-gov/stratos
-    tag_filter: '4.1.0'
+      #branch: master
+    branch: apb/rollback-stratos
 
 - name: next-stratos-release
```
Comment (apburnes) [inline ci/pipeline.yml:312]: Since we are no longer deploying stratos via releases, let's remove the `next-stratos-release` job.

**Review by apburnes — COMMENTED** [review]
Two lasts things for cleanup.

File: ci/pipeline.yml (hunk 5 above)
Comment (rcgottlieb) [reply ci/pipeline.yml:312]: I took care of both of these with my last commit.

File: ci/pipeline.yml [hunk 6]
```diff
@@ -266,24 +289,14 @@ resources:
...
-    tag_filter: '^(5\..*)'
+    branch: main
 
 - name: config
   type: git
   icon: github-circle
   source:
     uri: ((github-uri))
-    branch: ((github-branch))
+    branch: rg-deploy-stratos
```
Comment (apburnes) [inline ci/pipeline.yml:299]:
```suggestion
    branch: ((github-branch))
```

**Review by apburnes — APPROVED** [review]
Good to go!

**Pull request merged**